### PR TITLE
Update gesture handler and reanimated libraries

### LINKED ISSUE #4555: [Drag & Drop blocks] Upgrade dependencies related to gesture handling and animation
## Description

As part of the preparation work for the Drag&Drop project, we have to upgrade the following dependencies to use the newer APIs and benefit from enhancements of new versions.

- `react-native-gesture-handler`
- `react-native-reanimated`

**NOTE:** Most of this work was done in GB branch `rnmobile/try/drag-and-drop-blocks-hack-day-4`.

### PR DESCRIPTION
**Pull request by fluiddot**
**Gutenberg PR:** https://github.com/WordPress/gutenberg/pull/39098
**WordPress-Android PR:** https://github.com/wordpress-mobile/WordPress-Android/pull/16033
**WordPress-iOS PR:** https://github.com/wordpress-mobile/WordPress-iOS/pull/18065

To test:
Follow the testing instructions described in https://github.com/WordPress/gutenberg/pull/39098.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: third-party-podspecs/react-native-video.podspec.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
   "name": "react-native-video",
-  "version": "5.0.2-wp-2",
+  "version": "5.2.0-wp-2",
```
Comment (fluiddot) [inline third-party-podspecs/react-native-video.podspec.json:3]: Looks like `react-native-video` was recently updated ([reference](https://github.com/WordPress/gutenberg/pull/38426)) but its podspec wasn't updated.

**Review by geriux — APPROVED** [review]
LGTM! This aligns with the changes on Gutenberg [approved](https://github.com/WordPress/gutenberg/pull/39098) as well. 🚀

**Comment by fluiddot** [comment]
> LGTM! This aligns with the changes on Gutenberg [approved](https://github.com/WordPress/gutenberg/pull/39098) as well. 🚀

Thanks @geriux for reviewing the PRs 🙇! I've just referenced in the PR's description the WordPress-Android and WordPress-iOS PRs, in case you could take a look at them.

I noticed that the Android installable build is not working due to [this error](https://github.com/wordpress-mobile/WordPress-Android/pull/16033#issuecomment-1056835170), I'm currently investigating it but it's likely that it would require further changes in all PRs. I'll let you know when I apply the fix, thanks!

**Comment by fluiddot** [comment]
> I noticed that the Android installable build is not working due to [this error](https://github.com/wordpress-mobile/WordPress-Android/pull/16033#issuecomment-1056835170), I'm currently investigating it but it's likely that it would require further changes in all PRs. I'll let you know when I apply the fix, thanks!

The fix has been already applied, now both [WordPress-Android](https://github.com/wordpress-mobile/WordPress-Android/pull/16033) and [WordPress-iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/18065) are ready for review 🎊 .

**Pull request merged**